src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -281,10 +281,50 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     return this._loadPromise;
   }
 
+  /**
+   * handle decode error - reload the video and seek to last currentTime
+   * @returns {void}
+   */
+  handleDecodeError(): void {
+    NativeAdapter._logger.debug(`handleDecodeError currentTime <${this._videoElement.currentTime}>`);
+    this._playerTracks = this._getParsedTracks();
+    const prevCurrTime = this._videoElement.currentTime;
+    const prevActiveAudioTrack = this._getActivePKAudioTrack();
+    const prevActiveTextTrack = this._getActivePKTextTrack();
+
+    this._eventManager.removeAll();
+    this._loadPromise = null;
+
+    if (this._videoElement && this._videoElement.src) {
+      Utils.Dom.setAttribute(this._videoElement, 'src', '');
+      Utils.Dom.removeAttribute(this._videoElement, 'src');
+    }
+
+    this.load(prevCurrTime).then(() => {
+      this._eventManager.listenOnce(this._videoElement, Html5EventType.PLAYING, () => {
+        if (prevActiveAudioTrack) {
```

Comment:
is there a chance this is not always available? there is always an active audio(not like text track for example, where it could be actually off)